### ISSUE TITLE
captions: hiding captionTarget is not reflected in CC button

### ISSUE THREAD

**Issue by phloxic** [flowplayer/flash#256]
When the captionTarget is hidden by clicking on the CC button, the button changes to a red background.
Unfortunately there is no way to script this. But hiding the captionTarget plugin (content) does only half off the job from the user perspective: the background of the CC button is still light, and signals that captions should be visible.
There should be a toggle method to simulate _exactly_ what clicking on the CC button does, affecting both plugins.
Ideally this should be a caption plugin method as the plugin knows what its target is.

**Comment by gmontard** [comment]
:+1:

**Comment by sugerfunk** [comment]
So, I figured out how to fix this issue.  There are two issues to address.

1) The "display" keyword in the config does not affect the captionTarget plugin until after the caption plugin is already loaded.  Since it does not seem to initialize the display options (and therefore the _viewModel.visible property) until much later, it makes sense to add the following code to determine the button's state at the end of the parseIfLoadedAndViewAvailable() function in CaptionPlugin.as

```
var _viewModel:DisplayPluginModel;
_viewModel = _player.pluginRegistry.getPlugin(_config.captionTarget) as DisplayPluginModel;
if (_viewModel) { _captionView.button.isDown = _viewModel.visible; }
```

And of course, in order for that to work, the following getter needs to be added to CaptionViewDelegate.as:

```
internal function get button():CCButton { return _button; }
```

2) There needs to be a way to turn captions on and off via JavaScript which doesn't again result in the CC button's state being out of sync with the actual caption state, as proposed above.  This is relatively simple by copying the onClick code from the button itself and adding this external function to the CaptionPlugin.as class:

```
[External]
public function toggleCaptions():void {
    _captionView.button.isDown = _player.togglePlugin(_config.captionTarget);
}
```

Have a look, my friends! These changes worked like a charm for me! :-)

**Comment by danrossi** [comment]
It should be a dock button then it can be scripted and controlled more standard. that was on the cards but no scope to do so.

**Comment by danrossi** [comment]
and possibly a controlbar button also.

**Comment by sugerfunk** [comment]
I agree with you completely, but these changes will address current buggy behavior in the short term until the more scalable and extensible solution can be implemented.

**Comment by sugerfunk** [comment]
One more idea while we're on the subject... I think it makes sense to add a isShowing() function to the caption plugin which doesn't require JavaScript to catch up to the Flash state on the next cue point.  I used this code, but it may be possible to make it more concise:

```
    public function isShowing():Boolean {
        var _viewModel:DisplayPluginModel;
        _viewModel = _player.pluginRegistry.getPlugin(_config.captionTarget) as DisplayPluginModel;
        return _viewModel.visible;
    }
```
